### PR TITLE
Fix having a submodule as a primary source

### PR DESCRIPTION
**Pull request by marmarek**
Do not try to interpret Source0 tag to build automatic tarball name, if
it isn't going to be created. This is especially relevant when Source0
points at a submodule archive (created via "modules" entry in
.qubesbuilder).

TODO:
- [x] add a test (with a package actually having `Source0: @something@`)
- [x] verify if Debian and/or Arch doesn't need similar change

Fixes QubesOS/qubes-issues#9088

### CODE REVIEW

**Comment by fepitre** [comment]
Do you have an example for this case?

**Comment by marmarek** [comment]
yes: https://github.com/QubesOS/qubes-vmm-xen/pull/185

**Comment by fepitre** [comment]
Ok I see, so we need to improve the phase below create archive or what ever is needed to trigger archive creation if we provide necessary information from `.qubesbuilder`

**Comment by marmarek** [comment]
Here I used already existing functionality of `modules` entry in `.qubesbuilder` _instead of` standard create archive. The issue is, some parts of create archive were still active (especially building archive name), which failed when I used `@xen@` placeholder.

Alternative would be yet another type for `files` section - like when we have `url` now, maybe add `git-url`, and then have also commit-id/tag/branch + gpg key to verify (for tag or branch - commit id explicitly doesn't require anything extra). But since we support submodules, I thought it will be easier this way.

**Comment by fepitre** [comment]
> Alternative would be yet another type for `files` section - like when we have `url` now, maybe add `git-url`, and then have also commit-id/tag/branch + gpg key to verify (for tag or branch - commit id explicitly doesn't require anything extra). But since we support submodules, I thought it will be easier this way.

I like the idea, if entry is `git-url` instead of `url`, there we can add commit/branch/tag in the process, in that case we can keep other fields like pubkeys.

File: tests/test_cli.py [hunk 1]
```diff
@@ -191,6 +191,21 @@ def test_common_component_fetch(artifacts_dir):
         artifacts_dir
         / "distfiles/desktop-linux-xfce4-xfwm4/xfwm4-4.16.1.tar.bz2"
     ).exists()
+    assert (
+        artifacts_dir
+        / "distfiles/linux-gbulb/gbulb-0.6.3.tar.gz"
+    ).exists()
+    # verify files layout inside
+    subprocess.run(
```
Comment (fepitre) [inline tests/test_cli.py:199]: What do you verify? Is this looking at logs only?

Comment (fepitre) [reply tests/test_cli.py:199]: ok I missed the line that you look for README.rst

**Review by fepitre — APPROVED** [review]
-----BEGIN PGP SIGNED MESSAGE-----
Hash: SHA256

Approch HEAD being at commit c4ab9f6037cd5ddd9504d421092a920fabe40e3a
-----BEGIN PGP SIGNATURE-----

iQIzBAEBCAAdFiEEn6ZLkvlecGvyjiymSEAQtc3FduIFAmaZL/sACgkQSEAQtc3F
duJnWBAAhw1KEE3tqkgjm2v+jR0LEdGnj/pbc9QA+BxK7UOhATzSJXnkMdmsNWRf
P0x2FO8BRGmeKb6tDsHutumIulHgygrHohHzU9nfCXwTBbz8S0RrjMDBTveQ4wPE
1ppF+b83pOtKpHpC/cUWnEykEuVEb9HOyUTmwTyAaaf7LShHw41uzmJbwhc7Kltj
w9bV/sXOBmtIvBrC6nkKoXG49vY8w02sjgDvJkUVfIkW6AxObVIhe+oetqW4BsPZ
qjRhcct2M3/ZTRBEAuUr2+YJLQYq8hGEYPKeaGyigdpgg+ZnQnPrYww47/Y9M3H0
TUwf/ngXI3R6SQyh3Akvv8FimN9NIbklo59o29HS+v/mMXzUtSsVLheidmJVfpIp
zcx0wRELhHuedjnReqGEBxUD2ZYocckWuBcdnJDuYEZ8szvSHV7mwUkqfbEpN+FW
Nui8/ppOLazP6Z/XqWx5sbbsJVv4ERVd788iQmrE6altCvhZwhd+noRJBP+X0+fm
acxooEbcbi3AYOjF1jXYqlucb/TAv97OCr2bOmlXwjbTrKGdTqJ/USSX2vrLGhpN
o6KFaQp9PHBOXn2/dvn/3WxHfamdOti3wDdZvxZml6Kq2fUXsLFtKlVCV1gcLaei
faWJGsXObOC01IeKlrktYS+QJrV96jrZzioc2Pk+pJ03w0kASjo=
=/3qH
-----END PGP SIGNATURE-----

**Pull request merged**